velox/type/DecimalUtil.h

```diff
@@ -84,6 +84,35 @@ class DecimalUtil {
     }
   }
 
+  template <typename TOutput>
+  inline static std::optional<TOutput> rescaleBigint(
+      const int128_t inputValue,
+      const int toPrecision,
+      const int toScale,
+      const bool nullOnFailure) {
+    int128_t rescaledValue = inputValue;
+    bool isOverflow = __builtin_mul_overflow(
+        rescaledValue, DecimalUtil::kPowersOfTen[toScale], &rescaledValue);
+    // Check overflow.
+    if (rescaledValue < -DecimalUtil::kPowersOfTen[toPrecision] ||
+        rescaledValue > DecimalUtil::kPowersOfTen[toPrecision] || isOverflow) {
+      if (nullOnFailure) {
+        return std::nullopt;
+      }
+      VELOX_USER_FAIL(
+          "Cannot cast BIGINT '{}' to DECIMAL({},{})",
+          inputValue,
+          toPrecision,
+          toScale);
+    } else {
+      if constexpr (std::is_same_v<TOutput, UnscaledShortDecimal>) {
```

Comment:
I don't think `else` block is required here.

Would you please try to replace `if constexpr` with
```
return TOutput(rescaledValue);
```

Refer this https://github.com/facebookincubator/velox/blob/main/velox/functions/prestosql/DecimalArithmetic.cpp#L195

Reply:
Thanks for the feedback!
I removed the `else`.

Regarding replacing `if constexpr`, since I used https://github.com/facebookincubator/velox/blob/babb1c34f1653068fdb45a1a45c273fabb1fc43d/velox/type/DecimalUtil.h#L80, the input for `UnscaledShortDecimal` and `UnscaledLongDecimal` are different. Then it seems hard to rewrite it as `return TOutput(rescaledValue)`. Let me know what do you think.